fix(compiler): i18n - trim whitespace from i18n metadata

petebacondarwin opened the pull request:
Fixes #34147

AndrewKushnir commented on packages/compiler/src/i18n/extractor_merger.ts:
```diff
@@ -508,6 +508,8 @@ function _getI18nAttr(p: html.Element): html.Attribute|null {
 function _parseMessageMeta(i18n?: string): {meaning: string, description: string, id: string} {
   if (!i18n) return {meaning: '', description: '', id: ''};
 
+  i18n = i18n.trim();
```
I'm wondering if we should trim whitespace for `id` part only to avoid potentially breaking change in case there is a whitespace in front of the meaning part, i.e. for these messages below:
```
<div i18n="meaning|descripion">TEST</div>

<div i18n="
    meaning|descripion">TEST</div>
```
we'll generate different message ids (due to different `meaning` values). With the changes in this PR, we will generate the same message id for both, which is better, but may result in "new" messages to be translated in translation consoles.

petebacondarwin replied:
I wondered about this... I really don't like the idea that `meaning` is whitespace dependent in the templates.  It makes it very brittle (as indicated by your comment).
I appreciate that there is a danger of messing up people's translation files, if they have whitespace in their templates...

How about we make the new `$localize` message digest whitespace independent? This is not a breaking change since no one is using this yet. Then when people migrate from the legacy message ids, we transform the id to be whitespace agnostic?

AndrewKushnir replied:
@petebacondarwin making new `$localize` message digest whitespace independent sounds great 👍

AndrewKushnir reviewed (changes requested):
Looks great, but this change as is might lead to some minor breaking changes. So the proposal is to trim `id` part only.

AndrewKushnir reviewed (approved):
LGTM 👍

AndrewKushnir commented:
- [VE Presubmit](http://test/OCL:283407514:BASE:283407594:1575322785611:a02bfc3a)
- [Ivy Presubmit](https://test.corp.google.com/OCL:283407514:BASE:283408211:1575322952397:1856fc00)

petebacondarwin committed:
refactor(compiler): remove redundant `metaFromI18nMessage()` function

petebacondarwin committed:
fix(ivy): i18n - trim whitespace when parsing metadata

It is possible for HTML formatters to add whitespace
around the content of `i18n` attribute values. This can
make the meaning and custom ids brittle to simple
whitespace formatting.

This commit ensures that the metadata string extracted
from HTML `i18n` attributes is trimmed before being parsed
into meaning, description and custom id.

petebacondarwin committed:
fix(compiler): i18n - trim whitespace from i18n custom ids

Fixes #34147
